File: classes/DatabaseManager.php
<?php
class DatabaseManager{
    private object $pdo;
    public function __construct(){
        $dbConnection = Config::dbConnection();
        if($_SERVER['HTTP_HOST'] == "localhost"){
            $dsn = $dbConnection['dsn'];
            $user = $dbConnection['user'];
            $pass = $dbConnection['pass'];    
        }
        else{
            $dsn = $dbConnection['dsn'];
            $user = $dbConnection['user'];
            $pass = $dbConnection['pass'];  
        }
        $this->pdo = new PDO($dsn, $user, $pass, [
                PDO::ATTR_ERRMODE            => PDO::ERRMODE_EXCEPTION,
                PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC,
                PDO::ATTR_EMULATE_PREPARES   => false,
        ]);
        $this->pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

    } 
    public function fetchAll(string $sql,array $params = []):array{
        $stmt = $this->pdo->prepare($sql);
        $stmt->execute($params);
        return $stmt->fetchAll();
    }

    public function fetchOne(string $sql,array $params = []):?array{
        $stmt = $this->pdo->prepare($sql);
        $stmt->execute($params);
        $row = $stmt->fetch();
        return $row ?: null;
    }
    public function fetch(string $sql){
        $stmt = $this->pdo->prepare($sql);
        $stmt->execute();
        return $stmt->fetchAll(PDO::FETCH_ASSOC);
    }
    public function execute(string $sql, array $params = []): bool {
        $stmt = $this->pdo->prepare($sql);
        return $stmt->execute($params);
    }
}

File: python/DatabaseManager.py
import mysql.connector
from Config import Config
class DatabaseManager:
    def __init__(self):
        dbConnection = Config.connection_info()

        host = dbConnection['host']
        user = dbConnection['user']
        password = dbConnection['password']
        database = dbConnection['database']

        self.conn =mysql.connector.connect(
            host = host,
            user = user,
            password = password,
            database = database
        )
        self.cursor = self.conn.cursor(dictionary = True) # returns rows as dictionary

    def execute(self,sql,params=()):
        self.cursor.execute(sql,params)
        self.conn.commit()

    def fetchAll(self,sql,params=()):
        self.cursor.execute(sql, params)
        return self.cursor.fetchall()

    def fetch(self,sql):
        self.cursor.execute(sql)
        return self.cursor.fetchone()
    
    def fetchOne(self,sql,params=()):
        self.cursor.execute(sql,params)
        return self.cursor.fetchone()
    
    def close(self):
        self.cursor.close()
        self.conn.close()